pkg/remotedialer/client.go

```diff
@@ -27,11 +27,15 @@ func connectToProxy(proxyURL string, headers http.Header, auth ConnectAuthorizer
 	}
 	ws, resp, err := dialer.Dial(proxyURL, headers)
 	if err != nil {
```

Comment:
Should we move the defers up?

Reply:
could do a nil check on each and call it before the error checking? maybe over kill cause the response from Dial is usually `nil, nil, err`